src/Engine/Physics/Trigger.hpp

```diff
@@ -1,22 +1,17 @@
 #pragma once
 
-#include <btBulletCollisionCommon.h>
 #include <functional>
 #include <map>
-#include "ITrigger.hpp"
+#include <memory>
 
 class PhysicsManager;
 
-namespace Component {
-    class RigidBody;
-};
-
 namespace Physics {
-    class Shape;
+    class TriggerObserver;
 
     /// Represent a trigger that checks intersections of specific rigid bodies
     /// against itself.
-    class Trigger : public btCollisionWorld::ContactResultCallback, public ITrigger {
+    class Trigger {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') class 'Trigger' does not have a copy constructor which is recommended since the class contains a pointer to allocated memory. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AnoCopyConstructor)